di/direction.go

```diff
@@ -14,30 +18,52 @@ const (
 	DirectionBTT
 )
 
+const (
+	progression Direction = 1 << iota
+	// axisVertical is the bit for the axis, 0 for horizontal, 1 for vertical
+	axisVertical
+
+	// If this flag is set, the orientation is chosen
+	// using the [verticalSideways] flag.
+	// Otherwise, the segmenter will resolve the orientation based
+	// on unicode properties
+	verticalOrientationSet
+	// verticalSideways is set for 'sideways', unset for 'upright'
+	// It implies BVerticalOrientationSet is set
+	verticalSideways
+)
+
 // IsVertical returns whether d is laid out on a vertical
 // axis. If the return value is false, d is on the horizontal
 // axis.
-func (d Direction) IsVertical() bool {
-	return d == DirectionBTT || d == DirectionTTB
-}
+func (d Direction) IsVertical() bool { return d&axisVertical != 0 }
 
 // Axis returns the layout axis for d.
 func (d Direction) Axis() Axis {
-	switch d {
-	case DirectionBTT, DirectionTTB:
+	if d.IsVertical() {
 		return Vertical
-	default:
-		return Horizontal
 	}
+	return Horizontal
 }
 
+// SwitchAwis switch from horizontal to vertical (and vice versa), preserving
```

Comment:
SwitchAxis probably ;) The linter should have caught this, perhaps our CI needs a few improvements.
Possibly "switches" not "switch" as well if this line is being updated.

Reply:
Updated, thank you.